### ISSUE TITLE
[Crash] [TT 2091] [ME] Constant server crashes

### ISSUE THREAD

**Issue by darkdeath1332** [MinimallyCorrect/TickThreading#921]
Here is my log;

2013-07-21 11:57:18 [INFO] [TickThreading] Started TT metrics reporting. This can be disabled in PluginMetrics.cfg
2013-07-21 11:57:21 [INFO] [TickThreading] Skipping version tick handler c com.pahimar.ee3.core.handlers.VersionCheckTickHandler com.pahimar.ee3.core.handlers.VersionCheckTickHandler@7548fbb4
2013-07-21 11:57:36 [SEVERE] [TickThreading] TickThreading v1.0.0.2091 for MC1.4.7 on 1.4.7 tickthreading,mcpc,spigot,craftbukkit,forge,fml - 1.7.0_21-b11 is installed on this server!
If anything breaks, check if it is still broken without TickThreading
We don't want to annoy mod devs with issue reports caused by TickThreading.
Seriously, please don't.
If it's only broken with TickThreading, report it at https://github.com/nallar/TickThreading/issues/new

Also, you really should be making regular backups. (You should be doing that even when not using TT.)
2013-07-21 12:01:00 [SEVERE] [TickThreading] Removed c appeng.me.tile.TileColorlessCable @487104464 in Overworld/0 at x, y, z: -99666, 62, -99896 as it is invalid.
Found and fixed 2 discrepancies in tile/entity lists in Overworld/0
tiles - invalid: 1, missing: 0, duplicate: 0, unloaded: 0
entities -  missing: 1, duplicate: 0, unloaded: 1
Took 76ms
2013-07-21 12:02:37 [INFO] [TickThreading] Trying to unstick Entities in Overworld/0 - 
Found 0 threads, Killed 2 threads, Already stopped 10 threads.
2013-07-21 12:02:37 [INFO] [TickThreading] Trying to unstick World Tick - 
Found 0 threads, Killed 1 threads, Already stopped 7 threads.
2013-07-21 12:02:37 [SEVERE] [TickThreading] The server appears to have deadlocked.
Last tick 38s ago.
Waiting ThreadManagers: Entities in Overworld/0, World Tick

Definitely deadlocked: 
"Entities in Overworld/0 - 12" Id=159 BLOCKED on appeng.me.tile.TileController@55b0db0e owned by "Entities in Overworld/0 - 2" Id=149
    at appeng.me.tile.TileController.getCellArray(TileController.java:813)
    -  blocked on appeng.me.tile.TileController@55b0db0e
    at appeng.me.tile.TileController.requestUpdate(TileController.java:562)
    at appeng.me.GridInterface.requestUpdate(GridInterface.java:119)
    at appeng.me.tile.TileInterfaceBase.a(TileInterfaceBase.java:272)
    at cofh.core.CoreUtils.addToInventorySlot(CoreUtils.java:53)
    at thermalexpansion.core.Utils.addToAdjInventory(Utils.java:102)
    at thermalexpansion.transport.tileentity.TileTeleportItem.addToAdjInventory(TileTeleportItem.java:169)
    at thermalexpansion.transport.tileentity.TileTeleportItem.receiveItem(TileTeleportItem.java:131)
    at thermalexpansion.transport.tileentity.TileTeleportItem.sendItem(TileTeleportItem.java:107)
    at thermalexpansion.transport.tileentity.TileTeleportItem.a(TileTeleportItem.java:234)
    at appeng.util.inv.SidedInventoryWrapper.a(SidedInventoryWrapper.java:46)
    at appeng.util.inv.AdaptorIInventory.addItems(AdaptorIInventory.java:78)
    at appeng.me.tile.TileOutputCable.OutputItems(TileOutputCable.java:352)
    at appeng.me.tile.TileOutputCable.tickNetwork(TileOutputCable.java:369)
    at appeng.me.tile.TileController.g(TileController.java)
    -  locked appeng.me.tile.TileController@31b86805
    at nallar.tickthreading.minecraft.tickregion.TileEntityTickRegion.doTick(TileEntityTickRegion.java:90)
    at nallar.tickthreading.minecraft.tickregion.TickRegion.run(TickRegion.java:35)
    -  locked nallar.tickthreading.minecraft.tickregion.TileEntityTickRegion@8ccd652
    at nallar.tickthreading.minecraft.ThreadManager$2.run(ThreadManager.java:101)
    at nallar.tickthreading.minecraft.ThreadManager$1.run(ThreadManager.java:43)
    at nallar.tickthreading.util.FakeServerThread.run(FakeServerThread.java:16)

"Entities in Overworld/0 - 2" Id=149 BLOCKED on appeng.me.tile.TileController@31b86805 owned by "Entities in Overworld/0 - 12" Id=159
    at appeng.me.tile.TileController.getCellArray(TileController.java:813)
    -  blocked on appeng.me.tile.TileController@31b86805
    at appeng.me.tile.TileController.requestUpdate(TileController.java:562)
    at appeng.me.GridInterface.requestUpdate(GridInterface.java:119)
    at appeng.me.tile.TileInterfaceBase.a(TileInterfaceBase.java:272)
    at cofh.core.CoreUtils.addToInventorySlot(CoreUtils.java:53)
    at thermalexpansion.core.Utils.addToAdjInventory(Utils.java:102)
    at thermalexpansion.transport.tileentity.TileTeleportItem.addToAdjInventory(TileTeleportItem.java:169)
    at thermalexpansion.transport.tileentity.TileTeleportItem.receiveItem(TileTeleportItem.java:131)
    at thermalexpansion.transport.tileentity.TileTeleportItem.sendItem(TileTeleportItem.java:107)
    at thermalexpansion.transport.tileentity.TileTeleportItem.a(TileTeleportItem.java:234)
    at appeng.util.inv.SidedInventoryWrapper.a(SidedInventoryWrapper.java:46)
    at appeng.util.inv.AdaptorIInventory.addItems(AdaptorIInventory.java:78)
    at appeng.me.tile.TileOutputCable.OutputItems(TileOutputCable.java:352)
    at appeng.me.tile.TileOutputCable.tickNetwork(TileOutputCable.java:369)
    at appeng.me.tile.TileController.g(TileController.java)
    -  locked appeng.me.tile.TileController@55b0db0e
    at nallar.tickthreading.minecraft.tickregion.TileEntityTickRegion.doTick(TileEntityTickRegion.java:90)
    at nallar.tickthreading.minecraft.tickregion.TickRegion.run(TickRegion.java:35)
    -  locked nallar.tickthreading.minecraft.tickregion.TileEntityTickRegion@379bb087
    at nallar.tickthreading.minecraft.ThreadManager$2.run(ThreadManager.java:101)
    at nallar.tickthreading.minecraft.ThreadManager$1.run(ThreadManager.java:43)
    at nallar.tickthreading.util.FakeServerThread.run(FakeServerThread.java:16)

Attempting to recover without restarting.
2013-07-21 12:03:16 [INFO] [TickThreading] Trying to unstick Entities in Overworld/0 - 
Found 0 threads, Killed 2 threads, Already stopped 10 threads.
2013-07-21 12:03:16 [INFO] [TickThreading] Trying to unstick World Tick - 
Found 0 threads, Killed 1 threads, Already stopped 7 threads.
2013-07-21 12:03:16 [SEVERE] [TickThreading] The server appears to have deadlocked.
Last tick 38s ago.
Waiting ThreadManagers: Entities in Overworld/0, World Tick

Definitely deadlocked: 
"Entities in Overworld/0 - 12" Id=159 BLOCKED on appeng.me.tile.TileController@55b0db0e owned by "Entities in Overworld/0 - 2" Id=149
    at appeng.me.tile.TileController.getCellArray(TileController.java:813)
    -  blocked on appeng.me.tile.TileController@55b0db0e
    at appeng.me.tile.TileController.requestUpdate(TileController.java:562)
    at appeng.me.GridInterface.requestUpdate(GridInterface.java:119)
    at appeng.me.tile.TileInterfaceBase.a(TileInterfaceBase.java:272)
    at cofh.core.CoreUtils.addToInventorySlot(CoreUtils.java:53)
    at thermalexpansion.core.Utils.addToAdjInventory(Utils.java:102)
    at thermalexpansion.transport.tileentity.TileTeleportItem.addToAdjInventory(TileTeleportItem.java:169)
    at thermalexpansion.transport.tileentity.TileTeleportItem.receiveItem(TileTeleportItem.java:131)
    at thermalexpansion.transport.tileentity.TileTeleportItem.sendItem(TileTeleportItem.java:107)
    at thermalexpansion.transport.tileentity.TileTeleportItem.a(TileTeleportItem.java:234)
    at appeng.util.inv.SidedInventoryWrapper.a(SidedInventoryWrapper.java:46)
    at appeng.util.inv.AdaptorIInventory.addItems(AdaptorIInventory.java:78)
    at appeng.me.tile.TileOutputCable.OutputItems(TileOutputCable.java:352)
    at appeng.me.tile.TileOutputCable.tickNetwork(TileOutputCable.java:369)
    at appeng.me.tile.TileController.g(TileController.java)
    -  locked appeng.me.tile.TileController@31b86805
    at nallar.tickthreading.minecraft.tickregion.TileEntityTickRegion.doTick(TileEntityTickRegion.java:90)
    at nallar.tickthreading.minecraft.tickregion.TickRegion.run(TickRegion.java:35)
    -  locked nallar.tickthreading.minecraft.tickregion.TileEntityTickRegion@8ccd652
    at nallar.tickthreading.minecraft.ThreadManager$2.run(ThreadManager.java:101)
    at nallar.tickthreading.minecraft.ThreadManager$1.run(ThreadManager.java:43)
    at nallar.tickthreading.util.FakeServerThread.run(FakeServerThread.java:16)

"Entities in Overworld/0 - 2" Id=149 BLOCKED on appeng.me.tile.TileController@31b86805 owned by "Entities in Overworld/0 - 12" Id=159
    at appeng.me.tile.TileController.getCellArray(TileController.java:813)
    -  blocked on appeng.me.tile.TileController@31b86805
    at appeng.me.tile.TileController.requestUpdate(TileController.java:562)
    at appeng.me.GridInterface.requestUpdate(GridInterface.java:119)
    at appeng.me.tile.TileInterfaceBase.a(TileInterfaceBase.java:272)
    at cofh.core.CoreUtils.addToInventorySlot(CoreUtils.java:53)
    at thermalexpansion.core.Utils.addToAdjInventory(Utils.java:102)
    at thermalexpansion.transport.tileentity.TileTeleportItem.addToAdjInventory(TileTeleportItem.java:169)
    at thermalexpansion.transport.tileentity.TileTeleportItem.receiveItem(TileTeleportItem.java:131)
    at thermalexpansion.transport.tileentity.TileTeleportItem.sendItem(TileTeleportItem.java:107)
    at thermalexpansion.transport.tileentity.TileTeleportItem.a(TileTeleportItem.java:234)
    at appeng.util.inv.SidedInventoryWrapper.a(SidedInventoryWrapper.java:46)
    at appeng.util.inv.AdaptorIInventory.addItems(AdaptorIInventory.java:78)
    at appeng.me.tile.TileOutputCable.OutputItems(TileOutputCable.java:352)
    at appeng.me.tile.TileOutputCable.tickNetwork(TileOutputCable.java:369)
    at appeng.me.tile.TileController.g(TileController.java)
    -  locked appeng.me.tile.TileController@55b0db0e
    at nallar.tickthreading.minecraft.tickregion.TileEntityTickRegion.doTick(TileEntityTickRegion.java:90)
    at nallar.tickthreading.minecraft.tickregion.TickRegion.run(TickRegion.java:35)
    -  locked nallar.tickthreading.minecraft.tickregion.TileEntityTickRegion@379bb087
    at nallar.tickthreading.minecraft.ThreadManager$2.run(ThreadManager.java:101)
    at nallar.tickthreading.minecraft.ThreadManager$1.run(ThreadManager.java:43)
    at nallar.tickthreading.util.FakeServerThread.run(FakeServerThread.java:16)

Attempting to recover without restarting.

**Comment by darkdeath1332** [comment]
It keeps crashing every 2 minutes, please help.

**Comment by ghost** [comment]
Duplicate of issue #919.

**Comment by darkdeath1332** [comment]
And #921

**Comment by darkdeath1332** [comment]
I need to do this till its fixed..... I lost so many players... :(

```
@EventHandler
public void onSignClick(PlayerInteractEvent event){

    if (event.getAction() == Action.RIGHT_CLICK_BLOCK) {
        int d =event.getClickedBlock().getTypeId();

        if (d == 2510  || d == 2511 || d == 4461){

            event.setCancelled(true);
            Player player = event.getPlayer();
            player.sendMessage(ChatColor.RED + "THIS IS TEMPOLAIRY TILL THIS IS FIXED!");
            player.sendMessage("Im sorry but server keeps crashing due ME systems... See our report here; https://github.com/nallar/TickThreading/issues/921");
```
